### ISSUE TITLE
Question/Suggestion about Object.freeze with namespaces

### ISSUE THREAD

**Issue by bigtimebuddy** [rollup/rollup#1693]
Hello, I'm the release manager for [PixiJS](https://github.com/pixijs/pixi.js/) and we are moving to Rollup in our next major release. However, I've bumped into an issue related to the use of `Object.freeze` in builds. (related PR: #203).

For example:

```js
import * as utils from './utils';
export { utils };
```

This produces code that would look something like this:

```js
var utils = Object.freeze({
  foo: foo,
  bar: bar
});
exports.utils = utils;
```

In our use-case, we actually want to be able to _not_ freeze in order to mixin deprecations. This layer of compatibility provides a way for use to rename and move APIs without doing major bumps. By freezing the object, we are unable to provide a compatibility layer for deprecated exports, like this: 

```js
Object.defineProperty(PIXI.utils, 'baz', {
  get() {
    console.warn('PIXI.utils.baz has been moved to PIXI.utils.bar');
    return PIXI.utils.bar;
  }
});
// Throws: Cannot define property baz, object is not extensible
```

Our workaround involves importing each item explicitly, which is not super ideal or scalable.

```js
import { foo, bar } from './utils';
export const utils = { foo, bar };
```

Is there an alternative approach here?

If not, could freezing namespaces have an opt-out option? Maybe something similar to `legacy` flag (e.g., `freeze: false`, default is `true`) and/or possibly something similar to `external` list to opt-out to freeze for specific modules/namespaces (`freeze: ['./constants']`).

Thanks in advance for your help.

**Comment by bigtimebuddy** [comment]
Current workaround is to create a plugin to strip out `Object.freeze`:

```js
import MagicString from 'magic-string';

export default function thaw()
{
    const pattern = /Object.freeze\s*\(\s*([^)]*)\)/g;

    return {
        name: 'thaw',
        transformBundle(code)
        {
            const str = new MagicString(code);
            let hasReplacements = false;
            let match;

            while ((match = pattern.exec(code)))
            {
                hasReplacements = true;
                const start = match.index;

                str.overwrite(
                    start,
                    start + match[0].length,
                    match[1]
                );
            }

            if (!hasReplacements)
            {
                return null;
            }

            return {
                code: str.toString(),
                map: str.generateMap({ hires: true }),
            };
        },
    };
}
```

**Comment by lukastaegert** [comment]
Great you found a working solution. Did you try #1696, does it work for you? Currently, I am a little busy with the recent release but once things have settled there, I plan on looking at this in more detail.

**Comment by bigtimebuddy** [comment]
Yes, that PR also solves my issue. Thanks for considering this.

**Comment by lukastaegert** [comment]
Closed via #1696
